File: examples/maven-basic/src/main/java/example/StringFormatter.java
package example;

/**
 * Pure business-logic class with no boilerplate methods.
 * None of these methods should be matched by JaCoCo method filter rules,
 * so this class should NOT appear in verify output.
 *
 * Deliberately avoids: getters (get*), setters (set*), equals, hashCode, toString.
 */
public class StringFormatter {

    private final String prefix;
    private final String suffix;

    public StringFormatter(String prefix, String suffix) {
        this.prefix = prefix;
        this.suffix = suffix;
    }

    public String format(String value) {
        return prefix + value + suffix;
    }

    public String padLeft(String value, int width, char padChar) {
        if (value.length() >= width) {
            return value;
        }
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < width - value.length(); i++) {
            sb.append(padChar);
        }
        sb.append(value);
        return sb.toString();
    }

    public String padRight(String value, int width, char padChar) {
        if (value.length() >= width) {
            return value;
        }
        StringBuilder sb = new StringBuilder(value);
        for (int i = 0; i < width - value.length(); i++) {
            sb.append(padChar);
        }
        return sb.toString();
    }

    public String truncate(String value, int maxLength) {
        if (value.length() <= maxLength) {
            return value;
        }
        return value.substring(0, maxLength);
    }

    public String repeat(String value, int times) {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < times; i++) {
            sb.append(value);
        }
        return sb.toString();
    }
}

File: examples/sbt-basic/src/main/scala/example/StringFormatter.scala
package example

/**
 * Pure business-logic class with no boilerplate methods.
 * None of these methods should be matched by JaCoCo method filter rules,
 * so this class should NOT appear in verify output.
 */
class StringFormatter(prefix: String, suffix: String) {

  def format(value: String): String = s"$prefix$value$suffix"

  def padLeft(value: String, width: Int, padChar: Char = ' '): String = {
    if (value.length >= width) value
    else padChar.toString * (width - value.length) + value
  }

  def padRight(value: String, width: Int, padChar: Char = ' '): String = {
    if (value.length >= width) value
    else value + padChar.toString * (width - value.length)
  }

  def truncate(value: String, maxLength: Int): String = {
    if (value.length <= maxLength) value
    else value.substring(0, maxLength)
  }

  def repeat(value: String, times: Int): String = value * times
}
